plugins/Overlay/Overlay.php

```diff
@@ -37,4 +41,41 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = 'General_OverlayRowActionTooltipTitle';
         $translationKeys[] = 'General_OverlayRowActionTooltip';
     }
+
+    /**
+     * Returns if a request belongs to the Overlay page
+     *
+     * Whenever we change the Overlay, or any feature that is available on that page, this list needs to be adjusted
+     * Otherwise it can happen, that the session cookie is sent with samesite=lax, which might break the session in Overlay
+     * See https://github.com/matomo-org/matomo/pull/18648
+     */
+    public static function isOverlayRequest()
+    {
+        $module = Piwik::getModule();
+        $action = Piwik::getAction();
+        $method = Common::getRequestVar('method', '', 'string');
+
+        $isOverlay = $module == 'Overlay';
+        $referrerUrlQuery = parse_url(Url::getReferrer() ?? '', PHP_URL_QUERY);
+        parse_str($referrerUrlQuery, $referrerUrlQueryParams);
```

Comment:
not sure if it makes sense to use `UrlHelper::getArrayFromQueryString()` which we also use in other places? Probably not too important as it might do the same, just wondering for consistency

Reply:
I guess it doesn't do the exactly same for parameter arrays, but should be fine to use that one as well. will change that